### PR TITLE
feat: bonding discount

### PR DESCRIPTION
**Pull request by d0c0q**
- remove second argument of redeem function
- bonding discount

### CODE REVIEW

File: contracts/OtterPAWBondStakeDepository.sol [hunk 1]
```diff
@@ -0,0 +1,635 @@
...
+    bool public immutable isLiquidityBond; // LP and Reserve bonds are treated slightly different
+    address public immutable bondCalculator; // calculates value of LP tokens
+
+    address public staking; // to stake and claim if no staking warmup
+
+    Terms public terms; // stores terms for new bonds
+    Adjust public adjustment; // stores adjustment to BCV data
+
+    mapping(address => Bond) public bondInfo; // stores bond information for depositors
+    mapping(address => Discount[]) public discountInfo; // stores discount information for depositor
```
Comment (otterclamking) [inline contracts/OtterPAWBondStakeDepository.sol:66]: Why not put this in `bondInfo`?

File: contracts/OtterPAWBondStakeDepository.sol [hunk 2]
```diff
@@ -0,0 +1,635 @@
...
+        require(_staking != address(0));
+        staking = _staking;
+    }
+
+    /**
+     *  @notice set paw discount
+     *  @param _paw address
+     *  @param _discount uint
+     */
+    function setPAWDiscount(address _paw, uint256 _discount)
```
Comment (otterclamking) [inline contracts/OtterPAWBondStakeDepository.sol:244]: Need a `removePAWDiscount` function to disable it when necessary

Comment (d0c0q) [reply contracts/OtterPAWBondStakeDepository.sol:244]: Set discount to zero to remove discount

File: contracts/OtterPAWBondStakeDepository.sol [hunk 3]
```diff
@@ -0,0 +1,618 @@
...
+        Bond memory info = bondInfo[_recipient];
+        // (timestamp since last interaction / vesting term remaining)
+        uint256 percentVested = percentVestedFor(_recipient);
+
+        require(percentVested >= 10000, 'not fully vested'); // if fully vested
+        delete bondInfo[_recipient]; // delete user info
+        uint256 _amount = IsCLAM(sCLAM).balanceForGons(info.gonsPayout);
+        IERC20(sCLAM).transfer(_recipient, _amount); // pay user everything due
+        for (uint256 i = 0; i < discountInfo[_recipient].length; i++) {
+            IERC721(discountInfo[_recipient][i].paw).safeTransferFrom(
```
Comment (otterclamking) [inline contracts/OtterPAWBondStakeDepository.sol:365]: You can use `safeTransfer(_recipient, discountInfo[_recipient][i].tokenID)`

Comment (d0c0q) [reply contracts/OtterPAWBondStakeDepository.sol:365]: https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC721/ERC721.sol
https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC721/IERC721.sol

There's no safeTransfer function in ERC721.

**Review by otterclamking — APPROVED** [review]

**Pull request merged**